packages/enzyme/src/ReactWrapper.js

```diff
@@ -358,7 +358,8 @@ class ReactWrapper {
    * @returns {Boolean}
    */
   matchesElement(node) {
-    return this.single('matchesElement', () => nodeMatches(node, this.getNodeInternal(), (a, b) => a <= b));
+    const rstNode = getAdapter().elementToNode(node);
+    return this.single('matchesElement', () => nodeMatches(rstNode, this.getNodeInternal(), (a, b) => a <= b));
```

Comment:
let's do the elementToNode conversion here inside the arrow function, so that it happens after `this.single` succeeds.